src/components/datatable.component.ts

```diff
@@ -848,6 +842,12 @@ export class DatatableComponent implements OnInit, AfterViewInit, DoCheck {
    * @memberOf DatatableComponent
    */
   onBodyPage({offset}: any): void {
+    this.selected = [];
+    this.allRowsSelected = false;
+    this.select.emit({
```

Comment:
Beacuse we need to clear the selected rows on page change or sorting. The main goal of this is to select rows on the current page only.
Thanks.